katana/wui/core/static/core/js/main.js

```diff
@@ -132,6 +132,16 @@ var katana = {
 		$elem.siblings().removeClass('active');
 		$elem.addClass('active');
 		katana.$activeTab = katana.$view.find( '#' + uid ).removeClass('hidden');
+		var onSwitch = katana.$activeTab.find('[on-switch]');
```

Comment:
all of this can be replaced with katana.methodCaller( 'stringFunction', $elem )

Reply:
How does a coder know *when* to call this method?

Reply:
Here is a use case.  This will save a developer the onus of (re-)setting the global variable on every function. See the how to use below. 
![contextswitch1](https://user-images.githubusercontent.com/30508979/33671406-21f13194-da6d-11e7-9476-5a20825fb429.PNG)
![contextswitch2](https://user-images.githubusercontent.com/30508979/33671414-26ab3cde-da6d-11e7-93f2-71817ea88f13.PNG)

The user defined context switch function can hide popups, set variables or even set a timed function. It just has to be quick.